lib/gcloud/vision/project.rb

```diff
@@ -76,11 +83,146 @@ def self.default_project
           Gcloud::GCE.project_id
       end
 
+      ##
+      # Returns a new image from the given source.
+      #
+      # Image files sent to the Google Cloud Vision API should not exceed 4 MB.
+      # For multiple images in a request, the Vision API currently imposes an 8
+      # MB per request limit.
+      #
+      # Note that an object in Google Cloud Storage is a single entity;
+      # permissions affect only that object. "Directory permissions" do not
+      # exist (though default bucket permissions do exist). Make sure the code
+      # which performs your request has access to that image.
+      #
+      # @see https://cloud.google.com/vision/docs/image-best-practices Best
+      #   Practices
+      #
+      # @param [Gcloud::Storage::File, IO, StringIO, String] source a Google
```

Comment:
I think the most common types given here will be String (local path and gs URL), followed by File, followed by other IO types (Tempfile, StringIO) that respond to `#rewind` and `#read`, and then Gcloud::Storage::File. Would it make sense to order the types by likelihood?